.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,102 @@
+name: Build and push image
+
+on:
+  workflow_call:
+    inputs:
+      image:
+        description: "The desired name of the image to build"
+        default: 'pathfinder2'
+        required: false
+        type: string
+  workflow_dispatch:
+    inputs:
+      image:
+        description: "The desired name of the image to build"
+        default: 'pathfinder2'
+        required: false
+        type: string
+
+jobs:
+
+  build-and-push-image:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+      packages: write
+
+    steps:
+
+    -
+      name: Checkout repository
+      uses: actions/checkout@v3
+
+    -
+      name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        driver-opts: |
+          image=moby/buildkit:v0.10.6
+
+    -
+      name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: |
+          ${{ vars.GC_REGISTRY }}/${{ vars.GC_PROJECT_ID }}/${{ inputs.image }}
```

Comment:
Are these variables I need to set somewhere?

Reply:
same as above